### PR TITLE
DEVPROD-11578 Remove accidentally scheduled tests

### PR DESCRIPTION
**Pull request by SimonEismann**
**Jira Ticket:** [DEVPROD-11578](https://jira.mongodb.org/browse/DEVPROD-11578)

### What's Changed

This PR removes the tasks that were accidentally scheduled on unintended variants in [this PR](https://github.com/mongodb/genny/commit/85013fac53c260c2b1c669e526ec5c90175bb929#diff-68085041bd53721df02ca135080b495557837aad7a80acd63723835cc0777332R777).

### CODE REVIEW

**Review by blagoev — APPROVED** [review]

**Review by jawwadasghar — APPROVED** [review]
LGTM

**Review by thessem — APPROVED** [review]

File: evergreen.yml [hunk 1]
```diff
@@ -125,9 +125,8 @@ buildvariants:
 
   - name: validate-genny-tasks
     display_name: Validate Genny Tasks
-    modules: [dsi, PrivateWorkloads]
     run_on:
-      - amazon2-arm64
+      - amazon2-arm64-latest-small
```
Comment (SimonEismann) [inline evergreen.yml:129]: Swapped this to a more popular variant

File: evergreen.yml [hunk 2]
```diff
@@ -238,6 +237,16 @@ tasks:
...
+      - command: github.generate_token
+        params:
+          owner: 10gen
+          repo: dsi
+          expansion_name: dsi_x_access_token
+      - command: github.generate_token
+        params:
+          owner: 10gen
+          repo: PrivateWorkloads
+          expansion_name: private_workloads_x_access_token
```
Comment (SimonEismann) [inline evergreen.yml:249]: We currently use the same github app we use in SPS here. I've requested a new app for this purpose and will update the evergreen config accordingly once it is available.

**Comment by SimonEismann** [comment]
@johndaniels I had to modify the evergreen integration to always clone the git repos to avoid new commits in the DSI/PrivateWorkloads repos. Could you take another look at this since you reviewed the original code?

**Pull request merged**